test/framework/easyconfig.py

```diff
@@ -1204,6 +1204,32 @@ def test_dump_extra(self):
 
         dumped_ec = EasyConfig(testec)
 
+    def test_dump_template(self):
+        """ Test EasyConfig's dump() method for files containing templates"""
+        rawtxt = '\n'.join([
+            'easyblock = "EB_toy"',
+            '',
+            'name = "foo"',
+            'version = "0.0.1"',
+            '',
+            'homepage = "http://foo.com/"',
+            'description = "foo description"',
+            '',
+            'toolchain = {\'version\': \'dummy\', \'name\': \'dummy\'}',
+            '',
+            'sources = [\'%(namelower)s-%(version)s.TAR.gz\']',
```

Comment:
this probably relates to the `lower` above?

a file ending in `TAR.gz` is very different from a file ending in `tar.gz`; casing matters (we're not using Windows ;-))